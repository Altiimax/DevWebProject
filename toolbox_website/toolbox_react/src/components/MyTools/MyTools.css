#MyToolsCont {
  width: auto;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
  align-content: space-around;
}