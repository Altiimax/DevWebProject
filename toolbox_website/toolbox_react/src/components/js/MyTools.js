import React, { Component } from "react";

export default class MyTools extends Component {
  render() {
    return <div>Here will be displayed the tools.</div>;
  }
}
