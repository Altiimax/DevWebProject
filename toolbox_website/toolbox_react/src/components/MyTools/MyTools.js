import React, { useState } from "react";
import ReactDOM from "react-dom";
import { apiRequest } from "../../api/apiRequest.js";
import Tool from "../Tool/Tool.js";
import AddTools from "../AddTools/AddTools.js";
import "./MyTools.css";


function MyTools(props) {
  const [user_id] = useState(props.user_id);

  function getMyToolsApi(id_pers) {
    let endpoint = "/api/persons/" + id_pers + "/tools/";

    let req = new apiRequest();
    req.open("GET", `${endpoint}`);

    function displayPopup(){
      let p = <AddTools showPopUp={true}/>;
      ReactDOM.render(p, document.getElementById("MyToolsPopup"));
    }

    req.addEventListener("readystatechange", function () {
      if (this.readyState === 4) {
        if (this.status === 200) {
          let resp = JSON.parse(this.responseText);
          let key_y = 999;
          let toolList = [
            <button key={key_y} className="oneTool" id="addOneTool" onClick={displayPopup}>
              Add new tool
            </button>,
          ];
          
          for (let i in resp) {
            //pour les images il faudrait pouvoir les faire défiler onHover!!! Ici qu'une seule affichée..
            //il faut rajouter les images via le props 'img' j'ai des problèmes lors de l'import je sais pas pq!
            key_y++;
            toolList.push(
              <Tool 
                key={key_y}
                picture={resp[i].toolImages[0].image}
                name={resp[i].toolName}
                price={resp[i].toolPrice}
                desc={resp[i].toolDescription}
              />
            );
          }
          ReactDOM.render(toolList, document.getElementById("MyToolsCont"));
        }
      }
    });

    req.send();
  };

  document.getElementById("profileSection").innerHTML = "My Tools";
  getMyToolsApi(user_id);

  return (
    <>
    <div id="MyToolsCont"></div>
    <div id="MyToolsPopup"></div>
    </>
  );
}

export default MyTools;
