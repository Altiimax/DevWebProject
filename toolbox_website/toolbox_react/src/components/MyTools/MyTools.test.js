import React from "react";
import ReactDOM from "react-dom";
import MyTools from "./MyTools.js";
import TestRenderer from "react-test-renderer";
it("render without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<MyTools />, div);
});

/*
Test de la snapshot (a faire que si le component ne bouge pas trop dans le temps)
Ce test ajoute un dossier _snaphsot_dans lequel est contenu toutes les snapshots
*/
it("renders a table", () => {
  const testRenderer = TestRenderer.create(<MyTools />);
  const testInstance = testRenderer.toJSON();
  expect(testInstance).toMatchSnapshot();
});
