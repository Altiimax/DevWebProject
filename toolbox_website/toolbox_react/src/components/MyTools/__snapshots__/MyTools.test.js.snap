// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a table 1`] = `
<div>
  <table
    className="table table-dark table-striped table-bordered table-hover"
  >
    <thead>
      <tr>
        <th>
          Image
        </th>
        <th>
          Name
        </th>
        <th>
          Owner
        </th>
        <th>
          Renting infos
        </th>
        <th>
          State
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Image not found
        </td>
        <td>
          Screwdriver
        </td>
        <td>
          Bob
        </td>
        <td>
          Was borrowed on 20/04/2012 and must be given back before 24/05/2012
        </td>
        <td>
          Used
        </td>
      </tr>
      <tr>
        <td>
          Image not found
        </td>
        <td>
          Screwdriver
        </td>
        <td>
          Bob
        </td>
        <td>
          Was borrowed on 20/04/2012 and must be given back before 24/05/2012
        </td>
        <td>
          Used
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
